frontend/app/mappers/application-history-mapper.server.ts

```diff
@@ -0,0 +1,232 @@
+    confirm: relatedPerson?.ApplicantDetail.ConsentToSharePersonalInformationIndicator,
+    dateOfBirth: relatedPerson?.PersonBirthDate.dateTime,
+    firstName: relatedPerson?.PersonName.at(0)?.PersonGivenName.at(0),
+    lastName: relatedPerson?.PersonName.at(0)?.PersonSurName,
+    socialInsuranceNumber: relatedPerson?.PersonSINIdentification.IdentificationID,
+  };
+}
+
+interface ToPartnerResult {
+  confirm: boolean | undefined;
+  dateOfBirth: string | undefined;
+  firstName: string | undefined;
+  lastName: string | undefined;
+  socialInsuranceNumber: string | undefined;
+}
+
+interface ToRelatedPersonArgs {
+  ApplicantDetail: {
+    PrivateDentalInsuranceIndicator?: boolean | undefined;
+    InsurancePlan?: { InsurancePlanIdentification?: { IdentificationID?: string | undefined }[] | undefined }[] | undefined;
+    ConsentToSharePersonalInformationIndicator?: boolean | undefined;
+    AttestParentOrGuardianIndicator?: boolean | undefined;
+  };
+  PersonBirthDate: { dateTime: string };
+  PersonName: { PersonGivenName: string[]; PersonSurName: string }[];
+  PersonSINIdentification: { IdentificationID: string };
+  PersonRelationshipCode: { ReferenceDataName: string };
+}
+
+interface ToInsurancePlanArgs {
+  InsurancePlanIdentification?: { IdentificationID?: string | undefined }[] | undefined;
+}
+
+interface ToPersonalInfoArgs {
+  ApplicantDetail: {
+    PrivateDentalInsuranceIndicator?: boolean | undefined;
+    DisabilityTaxCreditIndicator?: boolean | undefined;
+    LivingIndependentlyIndicator?: boolean | undefined;
+    InsurancePlan?: { InsurancePlanIdentification?: { IdentificationID?: string | undefined }[] | undefined }[] | undefined;
+  };
+  PersonBirthDate: { dateTime: string };
+  PersonContactInformation: {
+    Address: {
+      AddressCategoryCode: { ReferenceDataName: string };
+      AddressCityName: string;
+      AddressCountry: { CountryCode: { ReferenceDataID: string } };
+      AddressPostalCode: string;
+      AddressProvince: { ProvinceCode: { ReferenceDataID: string } };
+      AddressSecondaryUnitText: string;
+      AddressStreet: { StreetName: string };
+    }[];
+    EmailAddress: { EmailAddressID: string }[];
+    TelephoneNumber: { TelephoneNumberCategoryCode: { ReferenceDataName: string; ReferenceDataID: string } }[];
+  }[];
+  PersonLanguage: { CommunicationCategoryCode: { ReferenceDataID: string }; PreferredIndicator: boolean }[];
+  PersonMaritalStatus: { StatusCode: { ReferenceDataID: string } };
+  PersonName: { PersonGivenName: string[]; PersonSurName: string }[];
+  PersonSINIdentification: { IdentificationID: string };
+  RelatedPerson: {
+    ApplicantDetail: {
+      PrivateDentalInsuranceIndicator?: boolean | undefined;
+      InsurancePlan?: { InsurancePlanIdentification?: { IdentificationID?: string | undefined }[] | undefined }[] | undefined;
+      ConsentToSharePersonalInformationIndicator?: boolean | undefined;
+      AttestParentOrGuardianIndicator?: boolean | undefined;
```

Comment:
We can remove the `undefined` keyword in favour of `?` - no need to have both:

(Same comment can be applied throughout this file where we have `someField? : string | undefined`

```suggestion
      AttestParentOrGuardianIndicator?: boolean;
```

Reply:
Fixed: a3363de42a431757737fa4c338f8126b2aca30e5